compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -314,6 +314,13 @@ declare_lint! {
     "imports that are never used"
 }
 
+declare_lint! {
+    /// [the reference]: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute
```

Comment:
You probably want to expand this before landing. There's examples above and below.